libqtile/backend/x11/drawer.py

```diff
@@ -163,11 +171,43 @@ def draw(
             0, 0,  # srcx, srcy
             offsetx, offsety,  # dstx, dsty
             self.width if width is None else width,
-            self.height if height is None else height
+            self.height if height is None else height,
         )
 
     def _find_root_visual(self):
         for i in self.qtile.core.conn.default_screen.allowed_depths:
             for v in i.visuals:
                 if v.visual_id == self.qtile.core.conn.default_screen.root_visual:
                     return v
+
+    def clear(self, colour):
+        # Some hacks needed for transparency
+
+        # Check if any colour is not fully opaque
+        if utils.has_transparency(colour):
+
+            # If the drawer isn't 24 bit then we need to do some extra work to clear it
+            # before drawing semi-opaque content
+            if not self.is_24_bit:
+
+                # RecordingSurface won't write clear operation to surface so we
+                # need to clear that surface directly.
+                if getattr(self, "_xcb_surface", None) is not None:
+                    ctx = cairocffi.Context(self._xcb_surface)
+                    ctx.save()
+                    ctx.set_operator(cairocffi.OPERATOR_CLEAR)
+                    ctx.paint()
+                    ctx.restore()
+
+                # May not be needed due to the above...
+                self.ctx.save()
+                self.ctx.set_operator(cairocffi.OPERATOR_CLEAR)
+                self.ctx.paint()
+                self.ctx.restore()
```

Comment:
With `background='ffffffff'` added to the default config's Bar I'm not seeing any transparency. This is with a plain red background:
![image](https://user-images.githubusercontent.com/44536113/121753863-245bb200-cb0b-11eb-8556-bbfbca4d5ee8.png)

Reply:
ffffffff is solid white, there should be no transparency with that.

Reply:
for alpha, "ff" is opaque and "00" is transparent.

Reply:
Lol yes of course